ddb_cp.module

```diff
@@ -243,19 +255,21 @@ function ddb_cp_ajax_test_result($site = NULL) {
   else {
     return MENU_NOT_FOUND;
   }
+
+  // @todo: Missing return statement it will note return any thing if the if statement is hit?
```

Comment:
If you call $json = ddb_cp_ajax_test_result(), you'll get the result MENU_NOT_FOUND.

If you call ddb_cp_ajax_test_result('dev') it'll print the latest Jenkins testreport in JSON format to stdout and finish off the request - and that's exactly what is needed for the Javascript code.

Go try for yourself: There's no hanging request if you call

https://wstest.ddbcms.dk/ajax/ddb_cp/test/result/dev

just a JSON-formatted Jenkins testreport.